app/master/build.py

```diff
@@ -99,7 +110,8 @@ def prepare(self, subjobs, job_config):
         :type subjobs: list[Subjob]
         :type job_config: JobConfig
         """
-        if self.project_type is None:
+        self._record_state_timestamp(BuildStatus.PREPARING)
```

Comment:
This wouldn't take into account the time it takes to fetch the project. Was that intentional? If you wanted to include the fetch times, we'd have to record this event in the build_request_handler in _build_preparation_loop().